src/nvim/normal.c

```diff
@@ -2939,6 +2939,9 @@ void check_visual_highlight(void)
  */
 void end_visual_mode(void)
 {
+  if (exiting) {
+    return;
+  }
```

Comment:
Oh wait, I didn't write how we got there. I remember that `end_visual_mode` was called after an exit event has been fired. At that point these resources where already freed. Should I throw that commit away and push it again, so you can take a closer look?